pentasharp/Services/TaxiReservationService.cs

```diff
@@ -0,0 +1,191 @@
+            try
+            {
+                _logger.LogInformation("Creating a new taxi reservation for user {UserId}.", model.UserId);
+
+                if (!TimeSpan.TryParse(model.ReservationTime, out var timeSpan))
+                {
+                    _logger.LogWarning("Invalid ReservationTime format: {ReservationTime}.", model.ReservationTime);
+                    return new BadRequestObjectResult(new { success = false, message = "Invalid ReservationTime format. Must be HH:mm:ss." });
+                }
+
+                var reservation = _mapper.Map<TaxiReservations>(model);
+                reservation.ReservationTime = model.ReservationDate.Add(timeSpan);
+                reservation.Status = ReservationStatus.Pending;
+                reservation.CreatedAt = DateTime.UtcNow;
+                reservation.Fare = 0;
+                _context.TaxiReservations.Add(reservation);
+                await _context.SaveChangesAsync();
+
+                _logger.LogInformation("Reservation created successfully with ID {ReservationId}.", reservation.ReservationId);
+                return new OkObjectResult(new { success = true, message = "Reservation created successfully." });
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occurred while creating a reservation.");
+                return new ObjectResult(new { success = false, message = "An internal server error occurred.", error = ex.Message })
+                {
+                    StatusCode = 500
+                };
+            }
+        }
+
+        public async Task<List<TaxiReservationRequest>> GetReservationsAsync()
+        {
+            try
+            {
+                _logger.LogInformation("Fetching all taxi reservations.");
+                var reservations = await _context.TaxiReservations
+                    .Include(r => r.User)
+                    .Include(r => r.Taxi)
+                    .Include(r => r.TaxiCompany)
+                    .ToListAsync();
+
+                var reservationInfo = _mapper.Map<List<TaxiReservationRequest>>(reservations);
+
+                foreach (var reservationDto in reservationInfo)
+                {
+                    var reservationEntity = reservations.FirstOrDefault(r => r.ReservationId == reservationDto.ReservationId);
+                    reservationDto.PassengerName = reservationEntity?.User?.FirstName ?? "Unknown";
+                    reservationDto.Driver = reservationEntity?.Taxi != null
+                        ? $"{reservationEntity.Taxi.DriverName} - {reservationEntity.Taxi.LicensePlate}"
+                        : "Unassigned";
+                }
+
+                _logger.LogInformation("Successfully fetched {Count} reservations.", reservationInfo.Count);
+                return reservationInfo;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occurred while fetching reservations.");
```

Comment:
can we fix the loggers to look something like:      
   _logger.LogError(ex, "Error occurred while processing reservations in {methodName}.", nameof(GetReservationsAsync));